test/regress/regress1/sets/choose4.smt2

```diff
@@ -0,0 +1,12 @@
+(set-logic ALL)
+(set-info :status sat)
+(set-option :produce-models true)
+(declare-fun A () (Set Int))
+(declare-fun a () Int)
+(assert (not (= A (as emptyset (Set Int)))))
+(assert (member 10 A))
+; this line raises an assertion error
+(assert (= a (choose A)))
```

Comment:
We need to the assertion failure thrown here.

Reply:
This is an assertion failure during --check-model you mean?

Reply:
This is related to the issue raised in [4175](https://github.com/CVC4/CVC4/pull/4175). See the results below:
```
./cvc4 choose1.smt2 
sat
(model
(define-fun A () (Set Int) (singleton 10))
(define-fun a () Int 10)
)
```
```
./cvc4 choose1.smt2 --check-models
Warning : SmtEngine::checkModel(): cannot check simplified assertion : (= (choice ((BOUND_VARIABLE_387 Int)) (ite (= A (as emptyset (Set Int))) (= BOUND_VARIABLE_387 (chooseUf271 A)) (and (member BOUND_VARIABLE_387 A) (= BOUND_VARIABLE_387 (chooseUf271 A))))) 10)
Warning : SmtEngine::checkModel(): cannot check simplified assertion : (= (choice ((BOUND_VARIABLE_398 Int)) (ite (= A (as emptyset (Set Int))) (= BOUND_VARIABLE_398 (chooseUf271 A)) (and (member BOUND_VARIABLE_398 A) (= BOUND_VARIABLE_398 (chooseUf271 A))))) 10)
Warning : SmtEngine::checkModel(): cannot check simplified assertion : (= (choice ((BOUND_VARIABLE_413 Int)) (ite (= A (as emptyset (Set Int))) (= BOUND_VARIABLE_413 (chooseUf271 A)) (and (member BOUND_VARIABLE_413 A) (= BOUND_VARIABLE_413 (chooseUf271 A))))) 10)
sat
(model
(define-fun A () (Set Int) (singleton 10))
(define-fun a () Int 10)
)
```
```
./cvc4 choose4.smt2 
sat
(model
(define-fun A () (Set Int) (singleton 10))
Fatal failure within CVC4::Expr CVC4::SmtEngine::getValue(const CVC4::Expr&) const at /home/mudathir/Desktop/CVC4/mudathirmahgoub/src/smt/smt_engine.cpp:2986
Check failure

 m->hasApproximations() || resultNode.getKind() == kind::LAMBDA || resultNode.isConst()

Aborted (core dumped)

```

Reply:
Can you add `--no-check-models` to the command line options for these regressions?  This is a common issue.

Reply:
Done. The tests pass for now. But I guess a permanent solution is needed for applications like alloy.